src/packet/tcp.rs

```diff
@@ -0,0 +1,197 @@
+        packet[IPV4_HEADER_LEN + TCP_MIN_HEADER_LEN + TCP_OPTIONS_LEN + 1] = 'e' as u8;
+        packet[IPV4_HEADER_LEN + TCP_MIN_HEADER_LEN + TCP_OPTIONS_LEN + 2] = 'o' as u8;
+        packet[IPV4_HEADER_LEN + TCP_MIN_HEADER_LEN + TCP_OPTIONS_LEN + 3] = 'w' as u8;
+        packet[IPV4_HEADER_LEN + TCP_MIN_HEADER_LEN + TCP_OPTIONS_LEN + 4] = '\n' as u8;
+
+        let ref_tcp_packet = [0xe0, 0x0f, // source port
+                              0x25, 0xc2, // dest port
+                              0xdf, 0x06, // sequence
+                              0xe6, 0xe6,
+                              0xc9, 0x2d, // acknowledgement
+                              0x98, 0x71,
+                              0x80,       // header length + reserved
+                              0x18,       // control bits
+                              0x02, 0xab, // window
+                              0xfe, 0x2d, // checksum
+                              0x00, 0x00, // urgent pointer
+                              0x01, 0x01, // tcp option no-op
+                              0x08, 0x0a, // tcp option timestamp, options length == 10 bytes
+                              0x1d, 0xfc, // 4-byte send timestamp
+                              0xcb, 0x76,
+                              0x1d, 0xfc, // 4-byte latest replied timestamp
+                              0xbe, 0x62,
+                              0x6d, 0x65, // payload
+                              0x6f, 0x77, 0x0a];
+        assert_eq!(&ref_tcp_packet[..], &packet[IPV4_HEADER_LEN..]);
+
+        //tcp/ipv4 == "45, 00 00 39 58 89 40 00 40 06 e4 33 7f 00 00 01 7f 00 00 01 e0 0f 25 c2 df 06 e6 e6 c9 2d 98 71 80 18 02 ab fe 2d 00 00 01 01 08 0a 1d fc cb 76 1d fc be 62 6d 65 6f 77 0a";
+
+    }
+
+    fn generate_simple_tcp_header<'p>(packet: &'p mut [u8]) {
+        let mut tcp_header = MutableTcpPacket::new(&mut packet[..]).unwrap();
+        tcp_header.set_source(57359);
+        assert_eq!(tcp_header.get_source(), 57359);
+
+        tcp_header.set_destination(9666);
+        assert_eq!(tcp_header.get_destination(), 9666);
+
+        tcp_header.set_sequence(3741771494);
+        assert_eq!(tcp_header.get_sequence(), 3741771494);
+
+        tcp_header.set_acknowledgement(3375208561);
+        assert_eq!(tcp_header.get_acknowledgement(), 3375208561);
+
+        tcp_header.set_data_offset_and_reserved(0x80);
+        assert_eq!(tcp_header.get_data_offset_and_reserved(), 0x80);
+
+        tcp_header.set_control_bits(0x18);
+        assert_eq!(tcp_header.get_control_bits(), 0x18);
+
+        tcp_header.set_window(0x02ab);
+        assert_eq!(tcp_header.get_window(), 0x02ab);
+
+        tcp_header.set_checksum(0xfe2d);
+        assert_eq!(tcp_header.get_checksum(), 0xfe2d);
+
+        tcp_header.set_urgent_pointer(0x0000);
+        assert_eq!(tcp_header.get_urgent_pointer(), 0x0000);
+
+        let tcp_option = TcpOptions{
+            kind: 0x1,
+            data: vec![0x01,0x08,0x0a,0x1d,0xfc,0xcb,0x76,0x1d,0xfc,0xbe,0x62],
```

Comment:
Yes I agree... this was temporarily raw. I'll write a high level abstraction for specifying the options.